code/game/objects/items/weapons/traps.dm

```diff
@@ -123,10 +124,552 @@
 	..()
 
 
-/obj/item/weapon/beartrap/update_icon()
+/obj/item/weapon/trap/update_icon()
+	icon_state = "[icon_base][deployed]"
+
+/obj/item/weapon/trap/animal
+	name = "small trap"
+	throw_speed = 2
+	throw_range = 1
+	gender = PLURAL
+	icon = 'icons/obj/items.dmi'
+	icon_base = "small"
+	icon_state = "small0"
+	desc = "A small mechanical trap thas is used to catch small animals like mice, lizards, chick and spiderlings."
+	throwforce = 2
+	force = 1
+	w_class = 2
+	origin_tech = list(TECH_MATERIAL = 1)
+	matter = list(DEFAULT_WALL_MATERIAL = 1750)
+	deployed = 0
+	time_to_escape = 3 // Minutes
+	var/breakout = FALSE
+	var/last_shake = 0
+	var/list/allowed_mobs = list(/mob/living/simple_animal/mouse, /mob/living/simple_animal/chick, /mob/living/simple_animal/lizard)
+	var/release_time = 0
+	var/list/resources = list(rods = 6)
+	var/spider = TRUE
+	health = 100
+	var/datum/weakref/captured = null
+
+/obj/item/weapon/trap/animal/examine(mob/user)
 	..()
+	if(captured)
+		var/datum/L = captured ? captured.resolve() : null
+		if (!L)
+			deployed = FALSE
```

Comment:
This is basically a partial state. And since we don't have a reset proc, it might be best to actually just, ignore this.
So just `return` should be sufficient?
```suggestion
```